core/src/main/java/tc/oc/pgm/listeners/PGMListener.java

```diff
@@ -271,7 +271,13 @@ public void lockTime(final MatchLoadEvent event) {
   @EventHandler
   public void unlockTime(final MatchStartEvent event) {
     boolean unlockTime = false;
-    if (!event.getMatch().getModule(WorldTimeModule.class).isTimeLocked()) {
+    Boolean isTimeLocked = event.getMatch().getModule(WorldTimeModule.class).isTimeLocked();
+    boolean isDaylightCycle =
+        Boolean.parseBoolean(event.getMatch().getWorld().getGameRuleValue(DO_DAYLIGHT_CYCLE));
+    // isTimeLocked() will be null if no timelock module is found
+    // if there is no timelock module and daylight cycle is enabled, unlock time
+    // if there is timelock module and its off, unlock time
+    if (((isTimeLocked == null) && isDaylightCycle) || (isTimeLocked != null && !isTimeLocked)) {
       unlockTime = true;
     }
```

Comment:
```java
Boolean isTimeLocked = event.getMatch().getModule(WorldTimeModule.class).isTimeLocked();
boolean isDaylightCycle =
    Boolean.parseBoolean(event.getMatch().getWorld().getGameRuleValue(DO_DAYLIGHT_CYCLE));
boolean unlockTime = isTimeLocked == null ? isDaylightCycle : !isTimeLocked;
```

This can be heavily simplified

However, if you look at the event on line 268, the gamerule is forced to false to prevent the match traversing while the map is loaded, i think queriying isDaylightCycle will always return false.

Check out what the behavior was prior to all of the time/world changes and keep it that way

Reply:
I have moved the isDayLightCycle boolean so it can be accessible to the first lockTime function, and record the gamerule before it locks the time. For some reason when I check the debugging it always seems to return true, even when I tested maps that have the gamerule to false like Moonlight Summit and Sunrise. This does not make a difference on what maps should have timelock though.